src/net/udp.rs

```diff
@@ -1,48 +1,61 @@
-use std::io;
-use std::net::{self, ToSocketAddrs};
+use std::io::{self, Cursor, ErrorKind, Error, Read, Write};
+use std::net::{self, SocketAddr, ToSocketAddrs};
 
-use super::{Packet, RawPacket, SocketState};
-use bincode::deserialize;
-
-use error::Result;
-
-const BUFFER_SIZE: usize = 1024;
+use super::{SocketState, NetworkConfig, constants};
+use packet::{header, Packet, PacketProcessor};
+use error::{NetworkError, Result};
 
 pub struct UdpSocket {
     socket: net::UdpSocket,
     state: SocketState,
-    recv_buffer: [u8; BUFFER_SIZE],
+    recv_buffer: [u8; constants::DEFAULT_MTU],
+    config: NetworkConfig,
+    packet_processor: PacketProcessor
 }
 
 impl UdpSocket {
     pub fn bind<A: ToSocketAddrs>(addr: A) -> io::Result<Self> {
         let socket = net::UdpSocket::bind(addr)?;
         let state = SocketState::new();
+        // TODO: add functionality to get config from file.
+        let config = NetworkConfig::default();
```

Comment:
I think we should just pass this in, let the user decide how to create the config.